### PR TITLE
Porting ocsp package from the latest standard lib

### PR DESCRIPTION
**Pull request by dissoupov**
A lot of OCSP responders return `400 Bad request` or `unauthorized`.
Building with the latest `ocsp` package from the standard lib, seems to be fixing the problem.

### CODE REVIEW

**Comment by dissoupov** [comment]
@dadrian @casebenton @zakird could you please take a look if it makes sense to backport standard `ocsp` package?
I see you did add this one 3 years ago and I'm not sure if there were some modifications in functionality, or it was simply copied.

Reason for the update:
- A lot of OCSP responders return 400 Bad request or unauthorized.
Building with the latest ocsp package from the standard lib, seems to be fixing the problem.
- The latest package allows to specify options for the request

@codyprime @bwireman FYI

File: x509/revocation/ocsp/ocsp.go [hunk 1]
```diff
@@ -1,173 +1,364 @@
...
+	case TryLater:
+		return "try later"
+	case SignatureRequired:
+		return "signature required"
+	case Unauthorized:
+		return "unauthorized"
+	default:
+		return "unknown OCSP status: " + strconv.Itoa(int(r))
 	}
-	return json.Marshal(&aux)
```
Comment (dadrian) [inline x509/revocation/ocsp/ocsp.go:51]: We vendored and edited it to add JSON support.

**Review by dadrian — COMMENTED** [review]
I'm fine updating the package, but the reason we backported it was to add JSON support. I don't know that anyone was using it, so if we have another way of doing JSON for OCSP, I'm fine (even prefer!) dropping the vendor.

File: x509/revocation/ocsp/ocsp.go (hunk 1 above)
Comment (dissoupov) [reply x509/revocation/ocsp/ocsp.go:51]: @dadrian I can try to restore JSON support if it's really needed.
what is the scenario for JSON in OCSP?

Comment (dadrian) [reply x509/revocation/ocsp/ocsp.go:51]: I believe it was done for data collection purposes for bulk revocation. I don't know that revocation ever came to fruition outside of some Censys prototypes, so it's probably fine to drop.

**Review by dadrian — APPROVED** [review]

**Pull request merged**